semgrep/semgrep/target_manager_extensions.py

```diff
@@ -13,47 +13,58 @@
 # and Language_util classes.
 # You may also have to regenerate some test snapshots with
 # pipenv run pytest tests/e2e/test_rule_parser.py --snapshot-update
-PYTHON_EXTENSIONS = [FileExtension(".py"), FileExtension(".pyi")]
-JAVASCRIPT_EXTENSIONS = [FileExtension(".js"), FileExtension(".jsx")]
-TYPESCRIPT_EXTENSIONS = [FileExtension(".ts"), FileExtension(".tsx")]
-JAVA_EXTENSIONS = [FileExtension(".java")]
+#
+# Extensions that are permissible for a given language. Semgrep-core will
+# perform its own filtering to check that the file is indeed in the correct
+# language before attempting full parsing. Additionally, all executable
+# files without an extension will also be passed to semgrep-core.
```

Comment:
What? Is this just for Bash? Should we restrict this to bash then?

Reply:
If this wasn't done before, it may be the reason for the slow downs. Perhaps we should not do it by default and require the user to explicitly enable it via a flag?

Reply:
> What? Is this just for Bash? Should we restrict this to bash then?

I'm trying to remove duplicate efforts, with a preference for ocaml over python.

This is for all the languages that can be run as scripts, which is about half of them. What would we be a benefit of doing this only when the requested language is bash?